internal/credit/postgresadapter/grant.go

```diff
@@ -109,18 +109,22 @@ func (g *grantDBADapter) ListGrants(ctx context.Context, params credit.ListGrant
 }
 
 func (g *grantDBADapter) ListActiveGrantsBetween(ctx context.Context, owner credit.NamespacedGrantOwner, from, to time.Time) ([]credit.Grant, error) {
+
 	query := g.db.Grant.Query().
-		Where(db_grant.And(db_grant.OwnerID(owner.ID), db_grant.Namespace(owner.Namespace))).
 		Where(
-			db_grant.Or(
-				db_grant.And(db_grant.EffectiveAtLT(from), db_grant.ExpiresAtGT(from)),
-				db_grant.And(db_grant.EffectiveAtGTE(from), db_grant.EffectiveAtLT(to)),
-				db_grant.EffectiveAt(from),
+			db_grant.And(
+				// Filter by owner, namespace
+				db_grant.OwnerID(owner.ID),
+				db_grant.Namespace(owner.Namespace),
+				// We are not interested in grants that either expired before the period
+				db_grant.Not(db_grant.EffectiveAtGT(to)),
+				// We are not interested in grants that are not yet effective
+				db_grant.Not(db_grant.ExpiresAtLT(from)),
+				// We are not interested in deleted or voided grants
+				db_grant.Or(db_grant.DeletedAtGTE(to), db_grant.DeletedAtIsNil()),
+				db_grant.Or(db_grant.VoidedAtGTE(to), db_grant.VoidedAtIsNil()),
```

Comment:
this query is quite tricky to get right (i've also rewritten it like 5 times by now), i'm gonna type the rules and then see if we agree in the query:
- this query has to return all grants that are active at ANY point in time in the queried period
- the period input can be nonsensical (e.g. from can be after to, that still has to work)
- the period can be a single point in time

Usually the periods we use are inclusive => exclusive (start => finish). In this case that's not quite the case, maybe the method could be renamed to `ListAllRelevantGrantsBetween`. We do have to list grants with deletedAt & voidedAt = to as otherwise they would never be accounted in any calculation:
Grant deleted (or voided) at t2
Query1: t1=>t2       (this should include those grants)
Query2: t2=>t3      (this should not include those grant)